### PR TITLE
Implemented user Authentication with Devise

### PR DESCRIPTION
**Pull request by cainCosmico**
This PR implements the devise authentication for rails. Added auth views and style with TailwindCSS .

- views with dark mode for Sign-in, sign up, edit profile, and forgot your password.
- Deploy on Heroku 
- Restrict views and controllers to unauthenticated users

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 class ArticlesController < ApplicationController
 
-  http_basic_authenticate_with name: "dhh", password: "secret", 
-                               except: [:index, :show]
+  # http_basic_authenticate_with name: "dhh", password: "secret",
```
Comment (estebancordoba) [inline app/controllers/articles_controller.rb:3]: remove unnecessary lines

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class CommentsController < ApplicationController
 
-  http_basic_authenticate_with name: "dhh", password: "secret", only: :destroy
+  # http_basic_authenticate_with name: "dhh", password: "secret", only: :destroy
```
Comment (estebancordoba) [inline app/controllers/comments_controller.rb:3]: remove unnecessary lines

File: app/javascript/stylesheets/application.scss [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      @apply bg-red-500 hover:bg-red-700 text-white font-bold py-2 px-4 rounded;
+    }
+    .nav_item {
+      @apply my-1 text-gray-100 hover:text-gray-200 md:mx-4 md:my-0;
+    }
+  }
+}
+
+
+// <div class="flex flex-wrap place-items-center h-screen">
```
Comment (estebancordoba) [inline app/javascript/stylesheets/application.scss:45]: same

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<p>Welcome <%= @email %>!</p>
+
+<p>You can confirm your account email through the link below:</p>
```
Comment (estebancordoba) [inline app/views/devise/mailer/confirmation_instructions.html.erb:3]: Did you implement a feature to confirm the email?

File: app/views/devise/mailer/reset_password_instructions.html.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<p>Hello <%= @resource.email %>!</p>
+
+<p>Someone has requested a link to change your password. You can do this through the link below.</p>
+
+<p><%= link_to 'Change my password', edit_password_url(@resource, reset_password_token: @token) %></p>
+
+<p>If you didn't request this, please ignore this email.</p>
```
Comment (estebancordoba) [inline app/views/devise/mailer/reset_password_instructions.html.erb:7]: Are these features implemented?

File: app/views/devise/shared/_links.html.erb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+<%- if devise_mapping.registerable? && controller_name != 'registrations' %>
+  <%= link_to "Sign up", new_registration_path(resource_name), class:"dark:text-gray-100" %><br />
+<% end %>
+
+<%- if devise_mapping.recoverable? && controller_name != 'passwords' && controller_name != 'registrations' %>
+  <%= link_to "Forgot your password?", new_password_path(resource_name), class:"dark:text-gray-100" %><br />
+<% end %>
+
+<%- if devise_mapping.confirmable? && controller_name != 'confirmations' %>
+  <%= link_to "Didn't receive confirmation instructions?", new_confirmation_path(resource_name), class:"dark:text-gray-100" %><br />
```
Comment (estebancordoba) [inline app/views/devise/shared/_links.html.erb:14]: How do you get these instructions?

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (cainCosmico) [reply app/controllers/comments_controller.rb:3]: commend removed

File: app/javascript/stylesheets/application.scss (hunk 3 above)
Comment (cainCosmico) [reply app/javascript/stylesheets/application.scss:45]: commend removed

File: app/views/devise/shared/_links.html.erb (hunk 6 above)
Comment (cainCosmico) [reply app/views/devise/shared/_links.html.erb:14]: The confirmation instructions are sent by email

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (cainCosmico) [reply app/controllers/articles_controller.rb:3]: removed

File: app/views/devise/mailer/reset_password_instructions.html.erb (hunk 5 above)
Comment (cainCosmico) [reply app/views/devise/mailer/reset_password_instructions.html.erb:7]: when a user makes a password recovery request, instructions are sent to the user's email address.

**Review by estebancordoba — APPROVED** [review]

**Pull request merged**